src/BaseModel.php

```diff
@@ -90,4 +114,11 @@ public function toArray()
 	{
 		return $this->data;
 	}
+
+	protected function checkValid()
+	{
+		if (empty($this->data['id'])) {
+			throw new HTTPException\InternalServerErrorException(static::class . ' record uninitialized');
```

Comment:
Really an exception? I'd prefer having it named `isValid()`, make it public so it can be checked if the model is valid at all (e.g. saved) and return a boolean. The `__get` should then throw the exception in case of "false".

The overall question is what to do with "new" model-entities, like we create a new "notify", which hasn't an ID at first? But this question is a disucssion in the dev-forum, because I'm currently stuck at excactly this situation for the enotify refactoring :-)

Reply:
Let's talk about it in the dev forum then.

This centralized check makes sense for the current state of models that can't exist outside of existing table rows. If we change this philosophy we'll be able to revisit this method.